File: exemplos/privacidade.java
class A {
  public void public_method(A x){
    x.x_private(x);
  }

  private void private_method(A x) {
    System.out.println("Private method");
  }
}

class Teste{
  public static void main(String[] args){
    A xpto1 = new A();
    A xpto2 = new A();
    xpto1.private_method(xpto1);
  }
}

File: exemplos/privacidade.rb
class A
  def public_method(a)
    a.private_method
  end

  private
  def private_method(d)
    puts "Privte method"
  end
end

xpto1 = A.new
xpto2 = A.new
xpto1.public_method(xpto2)
